sender.py

```diff
@@ -215,6 +216,7 @@ def send_tcp(self, encrypted):
             s.settimeout(TCP_TIMEOUT)   # generous timeout
             s.connect((self.address, self.port))
             s.sendall(encrypted)
+            self.received = s.recv(1024)
```

Comment:
Here's my suggestion:
```
received = s.recv(1024)
branch, flag = self.handle_return_packet(received)    # this function is basically extract_git_tag but return a bool instead of int for the flag
self.manager.branch = branch
self.quit_after_interval = flag
```